src/main/java/eu/xenit/gradle/docker/alfresco/tasks/AbstractWarEnrichmentTask.java

```diff
@@ -24,7 +25,10 @@
 
     protected AbstractWarEnrichmentTask() {
         outputWar.set(inputWar.flatMap(_x -> getProject().getLayout().getBuildDirectory()
-                .file("xenit-gradle-plugins/" + getName() + "/" + getName() + ".war")));
+                .file("xenit-gradle-plugins/" + getName() + "/" + getName() + ".war"))
+                .map(outputFile -> isEnabled() ? outputFile
```

Comment:
SInce I'm unfamiliar with the SDK, I am probably missing something, but won't this change result in _always_ throwing when on version < 6.2, while the intention is to only warn when you want to disable the task?

Reply:
What we do here is setting the `outputWar` property to `null` when this task is disabled.

This signals to the tasks that consume the output war (so `createDockerFile` & `alfrescoWar`) that there is no file to add.

We do not want to *warn* when disabling the task on < 6.2, we want to actively fail the build with a clear error message.

Returning `null` from a `map()` results in a vague exception in Gradle versions before 6.2. The `x ? y : z` operator is equivalent to `if(x) { return y;  }  else { return z; }`, the non-matching side will not be evaluated at all.

On the other hand, keeping the output file when the task is disabled will result in an exception from the tasks that consume the output war, because the promised output file does not actually exists (not generated, because the task is not run).

Reply:
Thank you for the thorough explanation.